src/Services/MediaService.php

```diff
@@ -79,14 +79,19 @@ public function first(array|int $where = []): ?Model
     public function createPreview(string $link): Model
     {
         $pathinfo = pathinfo($link);
+
         $link = Storage::path($pathinfo['basename']);
+        $name = "preview_{$pathinfo['basename']}";
+
+        $path = explode('/', $link);
+        array_pop($path);
+        $path = implode('/', $path) . "/{$name}";
```

Comment:
please try to use the `extract_last_part` helper instead

Reply:
Done.